[functorch] excise older custom_vjp prototype

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #84648
* #84642
* #84639
* __->__ #84638
* #84637

It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

Differential Revision: [D39315050](https://our.internmc.facebook.com/intern/diff/D39315050)

zou3519 committed:
[functorch] excise older custom_vjp prototype

It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39315050).

samdow commented on functorch/test/test_eager_transforms.py:
```diff
@@ -35,9 +34,6 @@
 from functorch.experimental import functionalize
 
-if not IS_WINDOWS:
-    from functorch._src.custom_function import custom_vjp
```
Do we want to get rid of "functorch._src.custom_function" too?

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

Differential Revision: [D39315050](https://our.internmc.facebook.com/intern/diff/D39315050)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39315050).

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

Differential Revision: [D39315050](https://our.internmc.facebook.com/intern/diff/D39315050)

[ghstack-poisoned]

zou3519 committed:
Update on "[functorch] excise older custom_vjp prototype"


It was based off of the Python op registration API that has been
implemented in PyTorch already, so we can always bring it back, but
we're probably taking a different approach here.

Test Plan:
- tests

Differential Revision: [D39315050](https://our.internmc.facebook.com/intern/diff/D39315050)

[ghstack-poisoned]
